### PR TITLE
Sections redesign

### PR DESCRIPTION
**Pull request by GuenoleK**
## API

I've updated the section service to sort and show the section in the case the user is connected or not. 
I've also updated the display of an article in a section in the case the user is connected or not.
# APP

I've redesigned the sections' display.

untitled screencast

### CODE REVIEW

**Comment by JabX** [comment]
Can you add a  `cursor:pointer` on the section items on the list (because they're clickable)?

File: api/src/services/section.ts [hunk 1]
```diff
@@ -5,53 +5,75 @@ import {fn, col} from 'sequelize';
...
+                const sectionAssociations = ((await ArticleSection.findAll()).map(association => association.get()));
+                let sectionsArray = [];
+                sectionAssociations.map(association =>
+                    articles.map(article => {
+                            if (article.id === association.ArticleId) {
+                                sections.map(section => {
+                                    if (section.id === association.SectionId) {
+                                        sectionsArray.push(section);
+                                    }
+                                });
```
Comment (JabX) [inline api/src/services/section.ts]: Identation looks weird here (nitpicking much)

File: app/src/back-office/views/layouts/home-layout.jsx [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
+@connect(
+    state => ({
+        connected: state.login.isConnected
+    }),
+    dispatch => ({
+        saveArticle: (title, description) => dispatch(saveArticle({title, description})),
+        showSnackBar: data => dispatch(showSnackBar(data))
+    })
+)
+@connect(state => ({snackBar: state.snackBar, connected: state.login.isConnected}))
```
Comment (JabX) [inline app/src/back-office/views/layouts/home-layout.jsx]: Hell no a single connect here please !

File: app/src/back-office/views/layouts/home-layout.jsx [hunk 3]
```diff
@@ -59,6 +99,24 @@ export default class HomeLayout extends Component {
...
+                    <i className='material-icons'>add</i>
+                </FloatingActionButton>
+            : null}
+
+                <Dialog
+                        title={i18n.t('article.create.dialog')}
+                        actions={[<FlatButton label={i18n.t('article.create.confirm')} primary={true} onClick={() => this.saveArticle()} />]}
+                        open={this.state.open}
+                        onRequestClose={() => this.toggleModal()}
+                    >
```
Comment (JabX) [inline app/src/back-office/views/layouts/home-layout.jsx:114]: Indentation

File: app/src/back-office/views/layouts/home-layout.jsx [hunk 4]
```diff
@@ -21,6 +31,36 @@ export default class HomeLayout extends Component {
...
+                    isError: true
+                });
+            }
+        } else {
+            this.props.showSnackBar({
+                message: 'edit-cartridge.content.snackBar.saveFailedContentMessage',
+                isError: true
+            });
+        }
+    }
```
Comment (JabX) [inline app/src/back-office/views/layouts/home-layout.jsx:58]: Why is this here? I believe we wanted to keep logic inside common as much as possible

File: app/src/common/components/section-list/index.jsx [hunk 5]
```diff
@@ -16,66 +13,78 @@ import i18n from 'i18next';
...
+                                <div className="panel">
+                                    <ArticleList />
+                                </div>
+                            </div>
+                        );
+                    }) : null
+                }
+            </div>
+        );
+    };
```
Comment (JabX) [inline app/src/common/components/section-list/index.jsx:49]: Did you have to do the accordion by hand? Material-UI doesn't have one?

Comment (GuenoleK) [inline app/src/common/components/section-list/index.jsx:49]: They got this : http://www.material-ui.com/#/components/list

I wanted to fit with this : https://support.google.com/chrome/?hl=fr#topic= @JabX

File: app/src/back-office/views/layouts/home-layout.jsx (hunk 4 above)
Comment (GuenoleK) [inline app/src/back-office/views/layouts/home-layout.jsx:58]: I've just done the exact same as the first layout you did. Now we have three diffrent layouts (one for the home, another one for the consult and one for the edit mode). If you have an idea to make it in common i'll follow you for the snackbar ;) @JabX
